roles/config_drive/molecule/default/converge.yml

```diff
@@ -54,3 +54,158 @@
+      register: _iso_check
+
+    - name: Assert ISO exists
+      ansible.builtin.assert:
+        that:
+          - _iso_check.stat.exists
+        fail_msg: "ISO file should exist after soft-clean scenario"
+        success_msg: "Soft-clean scenario passed - ISO exists and role didn't fail on skipped tasks"
+
+    - name: Run config_drive role again with same configuration (should pass - no changes)
+      vars:
+        cifmw_config_drive_uuid: de2f369a-1886-4a90-8e50-e419289e6850
+        cifmw_config_drive_name: test01
+        cifmw_config_drive_hostname: test01.example.com
+        cifmw_config_drive_userdata:
+          ssh_authorized_keys:
+            - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCg1LHRahLiT1NFv4l/XH
+          packages:
+            - git
+            - bind-utils
+          write_files:
+            - path: /root/test.file
+              owner: root:root
+              content: |-
+                # Test file content
+          runcmd:
+            - ['sh', '-c', 'echo foo | tee -a /tmp/foo']
+        cifmw_config_drive_networkconfig:
+          network:
+            version: 2
+            ethernets:
+              id0:
+                match:
+                  macaddress: "aa:bb:cc:dd:ee:ff"
+                addresses:
+                  - 192.168.0.101/24
+                routes:
+                  - to: 0.0.0.0/0
+                    via: 192.168.0.1
+                    on-link: true
+                nameservers:
+                  addresses:
+                    - 192.168.0.1
+      ansible.builtin.include_role:
+        name: config_drive
+
+    - name: Attempt to change user-data on existing ISO (should fail)
+      block:
+        - name: Include config_drive with modified user-data
+          vars:
+            cifmw_config_drive_uuid: de2f369a-1886-4a90-8e50-e419289e6850
+            cifmw_config_drive_name: test01
+            cifmw_config_drive_hostname: test01.example.com
+            cifmw_config_drive_userdata:
+              ssh_authorized_keys:
+                - ssh-rsa DIFFERENT_KEY_HERE
+              packages:
+                - vim  # Changed from git
+            cifmw_config_drive_networkconfig:
+              network:
+                version: 2
+                ethernets:
+                  id0:
+                    match:
+                      macaddress: "aa:bb:cc:dd:ee:ff"
+                    addresses:
+                      - 192.168.0.101/24
+                    routes:
+                      - to: 0.0.0.0/0
+                        via: 192.168.0.1
+                        on-link: true
+                    nameservers:
+                      addresses:
+                        - 192.168.0.1
+          ansible.builtin.include_role:
+            name: config_drive
+
+        - name: Fail if role didn't reject the change
```

Comment:
so if it does not fail earlier, definitely you should not trigger fail. It needs to fail in the config_drive role, otherwise the test does not make sense.

Reply:
This Fail task is there as a safety, if the config_drive role does not fail, as we expect it too do.
Without this the role could incorrectly not fail and the test would just pass without detecting the issue.

When the role fail this task is never executed, we end up in the "rescue" block below, whiuch will assert that the correct failure was raised.